eva_submission/eload_brokering.py

```diff
@@ -27,39 +29,46 @@ def __init__(self, eload_number: int, vcf_files: list = None, metadata_file: str
             if metadata_file:
                 self.eload_cfg.set('validation', 'valid', 'metadata_spreadsheet', value=os.path.abspath(metadata_file))
 
-    def broker(self):
+    def broker(self, brokering_tasks=None):
         """Run the brokering process"""
-        # Reset previous values that could have been set before
-        self.eload_cfg['brokering'] = {}
-        output_dir = self._run_brokering_prep_workflow()
-        self._collect_brokering_prep_results(output_dir)
-        shutil.rmtree(output_dir)
-
-        self.upload_to_bioSamples()
-        self.broker_to_ena()
-
-    def broker_to_ena(self):
-        ena_spreadsheet = os.path.join(self._get_dir('ena'), 'metadata_spreadsheet.xlsx')
-        self.update_metadata_from_config(self.eload_cfg['validation']['valid']['metadata_spreadsheet'], ena_spreadsheet)
-        submission_file, project_file, analysis_file = process_metadata_spreadsheet(ena_spreadsheet,
-                                                                                    self._get_dir('ena'), self.eload)
-        # Upload the VCF to ENA FTP
-        ena_uploader = ENAUploader(self.eload)
-        files_to_upload = [vcf_file for vcf_file in self.eload_cfg['brokering']['vcf_files']] + \
-                          [self.eload_cfg['brokering']['vcf_files'][vcf_file]['index'] for vcf_file in self.eload_cfg['brokering']['vcf_files']]
-        ena_uploader.upload_vcf_files_to_ena_ftp(files_to_upload)
-
-        # Upload XML to ENA
-        ena_uploader.upload_xml_files_to_ena(submission_file, project_file, analysis_file)
-
-        self.eload_cfg.set('brokering', 'ena', value=ena_uploader.results)
-
-    def upload_to_bioSamples(self):
+        self.prepare_brokering(force='preparation' in brokering_tasks)
+        self.upload_to_bioSamples(force='biosamples' in brokering_tasks)
+        self.broker_to_ena(force='ena' in brokering_tasks)
```

Comment:
this might be just me but i kept reading these as setting `force` to a string rather than a boolean...  you could do `force=('ena' in brokering_tasks)` if you want to humor me i guess :)

Reply:
You're right it is a bit confusing. I've changed it to your suggestion.